### PR TITLE
Fix serve-d being stuck in endless loop after closing vscode

### PR DESCRIPTION
**Pull request by ryuukk**
That should be enough to fix: 

- https://github.com/Pure-D/serve-d/issues/185
- https://github.com/Pure-D/serve-d/issues/186

Although in my testing dcd-server stayed alive, but at 0% cpu usage, so.. i'm not sure where to check right now

I also edited this PR from the github website, code style might get altered, sorry about that!

### CODE REVIEW

**Comment by ryuukk** [comment]
Hmm, actually that might not be enough, it's random, sometimes process closes, sometimes it stays...

I'm not sure anymore.. sorry!

File: source/workspaced/com/dcd.d [hunk 1]
```diff
@@ -228,15 +228,32 @@ class DCDComponent : ComponentWrapper
...
-				while (serverPipes.stderr.isOpen && !serverPipes.stderr.eof)
+			{
+				try
+				{
+					while (running)
+					{
+						auto line = serverPipes.stderr.readln();
+						if (line !is null)
+							trace("Server: ", line); // evaluates lazily, so read before
+						else
```
Comment (WebFreak001) [inline source/workspaced/com/dcd.d:245]: hm is this guaranteed to work though? The phobos documentation says nothing about readln ever returning null or it meaning it's no longer running.

I think readln definitely returns nothing when the File is eof or no longer opened, which the previous while checked.

Comment (ryuukk) [reply source/workspaced/com/dcd.d:245]: io errors should be handled via the exception it throws anyways, the ``!is null`` follows the example in the doc..

It's a pretty complex bug to track..

**Comment by ryuukk** [comment]
I sent a PR on serve-d project, so it at least doesn't consume 100% of the cpu..

But the problem remains, the process doesn't close.. something is wrong somewhere but i don't know where..

When i try to debug, in debug build, the process closes, but not in release.. this makes it hard to debug

I personally think the code is way too complicated.. it needs to be redone to make it simpler without the mixins.. they make the code very hard to debug unfortunatly..

**Comment by WebFreak001** [comment]
merged it with a custom commit, don't think I can mark it as merged in github

**Pull request closed without merging**